luarules/gadgets/game_initial_spawn.lua

```diff
@@ -114,6 +118,220 @@ if gadgetHandler:IsSyncedCode() then
+			return teamPlayerData[teamID].skill
+		else
+			return 0
+		end
+	end
+
+	local function GetSkillByPlayer(playerID)
+		if playerID == nil then return 0 end -- uhh
+		local customtable = select(11, Spring.GetPlayerInfo(playerID))
+		if type(customtable) == 'table' then
+			local tsMu = customtable.skill
+			local tsSigma = customtable.skilluncertainty
+			local ts = tsMu and tonumber(tsMu:match("%d+%.?%d*"))
+			if (ts == nil) then return 0 else return ts end
+		end
+		return 0
+	end
+
+	local function compareSkills(teamA, teamB)
+		local skillA = GetSkillByTeam(teamA)
+		local skillB = GetSkillByTeam(teamB)
+		return skillA > skillB  -- Sort in descending order of skill
+	end
+
+	local function isAllyTeamSkillZero(allyTeamID)
+		local teams = Spring.GetTeamList(allyTeamID)
+		for _, teamID in ipairs(teams) do
+			local _, _, _, isAiTeam = Spring.GetTeamInfo(teamID)
+			if not isAiTeam and gaiaAllyTeamID ~= allyTeamID then
+				local skill = GetSkillByTeam(teamID)
+				if skill ~= 0 then
+					return false
+				end
+			end
+		end
+		return true
+	end
+
+	local function sendTeamOrder(teamOrder, allyTeamID_ready, log)
+		-- we send to allyTeamID Turn1 Turn2 Turn3 {...}
+		local orderMsg = ""
+		local orderIds = ""
+		local alone = true
+		for i, teamID in ipairs(teamOrder) do
+			local tname = teamPlayerData[teamID].name or "unknown" -- "unknown" should not happen if we create the order after everyone connects, so we are good
+			if i == 1 then
+				orderMsg = tname
+				orderIds = FindPlayerIDFromTeamID(teamID)
+			else
+				if alone then alone = false end
+				orderMsg = orderMsg .. ", " .. tname
+				orderIds = orderIds .. " " .. FindPlayerIDFromTeamID(teamID)
+			end
+		end
+		if log and not alone then
+			Spring.Log(gadget:GetInfo().name, LOG.INFO, "Order [id:"..allyTeamID_ready.."]: "..orderMsg)
+		end
+		Spring.SendLuaUIMsg("DraftOrderPlayersOrder " .. allyTeamID_ready .. " " .. orderIds)
+	end
+
+	local function calculateVotedPercentage(allyTeamID)
+		local totalPlayers = Spring.GetTeamCount(allyTeamID)
+		local votedPlayers = 0
+		for teamID, _ in pairs(votedToForceSkipTurn[allyTeamID]) do
+			if votedToForceSkipTurn[allyTeamID][teamID] then
+				votedPlayers = votedPlayers + 1
+			end
+		end
+		return (votedPlayers / totalPlayers) * 100
+	end
+
+	local function checkVotesAndAdvanceTurn(allyTeamID)
+		if allyTeamSpawnOrderPlaced[allyTeamID] >= #allyTeamSpawnOrder[allyTeamID] then return end
+		local votedPercentage = calculateVotedPercentage(allyTeamID)
+		if votedPercentage > 50 then
+			--Spring.Echo("Over 50% of players on ally team " .. allyTeamID .. " have voted.")
+			for teamID, _ in pairs(votedToForceSkipTurn[allyTeamID]) do
+				votedToForceSkipTurn[allyTeamID][teamID] = false -- reset vote
+			end -- and advance turn
+			allyTeamSpawnOrderPlaced[allyTeamID] = allyTeamSpawnOrderPlaced[allyTeamID]+1
+			SendDraftMessageToPlayer(allyTeamID, allyTeamSpawnOrderPlaced[allyTeamID])
```

Comment:
`SendDraftMessageToPlayer` is getting called before it is initialized

Reply:
There a if check before this code is reached. both "skip_my_turn" and "vote_skip_turn" checks if the queue has data before this code can be reached in the first place.
```
					if allyTeamSpawnOrderPlaced[allyTeamID] and allyTeamSpawnOrderPlaced[allyTeamID] > 0 then
						if msg == "skip_my_turn" then
```